vendor/parser-extended/parsetree.mli

```diff
@@ -862,7 +862,7 @@ and module_type =
 and module_type_desc =
   | Pmty_ident of Longident.t loc  (** [Pmty_ident(S)] represents [S] *)
   | Pmty_signature of signature  (** [sig ... end] *)
-  | Pmty_functor of functor_parameter loc list * module_type
+  | Pmty_functor of syntax_kind * functor_parameter loc list * module_type
       (** [functor (X1 : MT1) ... (Xn : MTn) -> MT] *)
```

Comment:
It's either `functor (X1 : MT1) ... ->` or `() ->`. I think this would be better represented with a new constructor `Pmty_gen of module_type`.

Reply:
done